Add "no-issue-needed" rule directly in CONTRIBUTING.rst (#23802)
mistercrunch/Airflow: CONTRIBUTING.rst

Issue reporting and resolution process
--------------------------------------

An unusual element of the Apache Airflow project is that you can open a PR to
fix an issue or make an enhancement, without needing to open an issue first.
This is intended to make it as easy as possible to contribute to the project.

If you however feel the need to open an issue (usually a bug or feature request)

> consider first to start `GitHub Discussion <https://github.com/apache/airflow/discussions>`_.

Reviewer: ```suggestion
consider starting with a `GitHub Discussion <https://github.com/apache/airflow/discussions>`_ instead.
```

> In vast majority of cases, discussions are better than issues - you should only open

Reviewer: ```suggestion
In the vast majority of cases discussions are better than issues - you should only open
```

> issues if you are sure you found a bug and have reproducible case to make the
> maintainers aware of it, or when you want to raise a feature request which you think
> does not require a lot of discussion. In case you have a very important topic
> to discuss - start a discussion on the `Devlist <https://lists.apache.org/list.html?dev@airflow.apache.org>`_.

Reviewer: ```suggestion
issues if you are sure you found a bug and have a reproducible case,
or when you want to raise a feature request that will not require a lot of discussion.
If you have a very important topic to discuss, start a discussion on the
`Devlist <https://lists.apache.org/list.html?dev@airflow.apache.org>`_ instead.
```
